packages/sourcegraph-extension-api/src/sourcegraph.d.ts

```diff
@@ -924,6 +924,43 @@ declare module 'sourcegraph' {
         Success = 5,
     }
 
+    /**
+     * A style for {@link BadgeDecorationAttachmentRenderOptions}.
+     */
+    export interface ThemableBadgeDecorationAttachmentStyle {
+        /** The icon (a base64-encoded PNG icon) to display next to the wrapped value. */
+        icon?: string
+
+        /** The CSS background-color property value for the attachment. */
+        backgroundColor?: string
+
+        /** The CSS color property value for the attachment. */
+        color?: string
+    }
+
+    /** A decoration attachment adds content to a hover tooltip or result in a locations panel. */
+    export interface BadgeDecorationAttachmentRenderOptions extends ThemableBadgeDecorationAttachmentStyle {
```

Comment:
A decoration is attached to a line. Let's just call this a `Badge`.

```suggestion
    export interface BadgeAttachmentRenderOptions extends ThemableBadgeDecorationAttachmentStyle {
```

Reply:
Updated.